ambari-server/src/main/resources/stacks/stack_advisor.py

```diff
@@ -1679,6 +1741,59 @@ def recommendConfigurationsForSSO(self, services, hosts):
 
     return recommendations
 
+  def recommendConfigurationsForKerberos(self, services, hosts):
+    self.services = services
+
+    stackName = services["Versions"]["stack_name"]
+    stackVersion = services["Versions"]["stack_version"]
+    hostsList = [host["Hosts"]["host_name"] for host in hosts["items"]]
+    servicesList = [service["StackServices"]["service_name"] for service in services["services"]]
+    components = [component["StackServiceComponents"]["component_name"]
```

Comment:
no, no, no...if you have such huge construction which is hardly readable with 2-cycles inside.....just use simple `for`  generators generally r slower, while `for` is bit faster and more readable.   the generators above r fine

Reply:
This is copied code from `recommendConfigurations`...  I will attempt to fix.